test/e2e_node/pod_resize_test.go

```diff
@@ -307,56 +309,164 @@ func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
+			accumulateContainerValues()
+			continue
+		}
+
+		var expectedCPUShares int64
+		var expectedCPULimitString, expectedMemLimitString string
+		expectedMemLimitInBytes = tc.Resources.Limits.Memory().Value()
+		cpuRequest = tc.Resources.Requests.Cpu()
+		cpuLimit = tc.Resources.Limits.Cpu()
+		if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+			expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+		} else {
+			expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+		}
+		cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+		if cpuLimit.IsZero() {
+			cpuQuota = -1
+		}
+		expectedCPULimitString = strconv.FormatInt(cpuQuota, 10)
+		expectedMemLimitString = strconv.FormatInt(expectedMemLimitInBytes, 10)
+		if podOnCgroupv2Node {
+			if expectedCPULimitString == "-1" {
+				expectedCPULimitString = "max"
+			}
+			expectedCPULimitString = fmt.Sprintf("%s %s", expectedCPULimitString, CPUPeriod)
+			if expectedMemLimitString == "0" {
+				expectedMemLimitString = "max"
 			}
-			err = verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10))
+			// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+			// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+			expectedCPUShares = 1 + ((expectedCPUShares-2)*9999)/262142
+		}
+		if expectedMemLimitString != "0" {
+			err := verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString)
 			if err != nil {
 				return err
 			}
 		}
+		err := verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString)
+		if err != nil {
+			return err
+		}
+		err = verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10))
+		if err != nil {
+			return err
+		}
+
+		accumulateContainerValues()
+	}
+
+	if !IsCgroup2UnifiedMode() {
+		// cgroup v1 is in maintenance mode. Skip verifying pod cgroup
+		return nil
+	}
+
+	// Verify pod cgroup
+
+	verifyPodCgroupValue := func(cgPath, expectedCgValue string) error {
+		cgValueBytes, err := os.ReadFile(cgPath)
+		if err != nil {
+			return fmt.Errorf("failed to read pod cgroup value %q", cgPath)
+		}
+		cgValue := strings.Trim(string(cgValueBytes), "\n")
+		if cgValue != expectedCgValue {
+			return fmt.Errorf("pod cgroup value '%s' not equal to expected '%s'", cgValue, expectedCgValue)
+		}
+		return nil
+
+	}
+
+	expectedPodCPUShares := int64(kubecm.MilliCPUToShares(podCPURequestMilliValue))
+	expectedPodCPUQuota := int64(-1)
+	if podCPULimitMilliValue > 0 {
+		expectedPodCPUQuota = kubecm.MilliCPUToQuota(podCPULimitMilliValue, kubecm.QuotaPeriod)
+	}
+	expectedPodCPULimitString := strconv.FormatInt(expectedPodCPUQuota, 10)
+	if podMemoryLimitInBytes < 0 {
+		podMemoryLimitInBytes = 0
+	}
+	expectedPodMemLimitString := strconv.FormatInt(podMemoryLimitInBytes, 10)
+	if podOnCgroupv2Node {
```

Comment:
This `if` can be removed. Here we're sure that `podOnCgroupv2Node == true`.

Reply:
Thanks.